src/main/java/com/bushidowallet/core/bitcoin/ecdsa/ECDSASignature.java

```diff
@@ -1,74 +1,98 @@
-
-        return writer.toBytes();
-    }
-
-    //DER signature format: [30] [total len] [02] [len R] [R] [02] [len S] [S]
-    public byte[] toDER() {
-        return toBytes();
-    }
-}
-
+package com.bushidowallet.core.bitcoin.ecdsa;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.math.BigInteger;
+
+import org.bouncycastle.asn1.DERInteger;
+import org.bouncycastle.asn1.DERSequenceGenerator;
+
+import com.bushidowallet.core.bitcoin.util.ByteWriter;
+
+/**
+ * Created by Jesion on 2015-04-11.
+ */
+public class ECDSASignature {
+
+    public BigInteger r;
+
+    public BigInteger s;
+
+    public boolean compressed;
+
+    public ECDSASignature(BigInteger r, BigInteger s, boolean compressed) {
+        this.r = r;
+        this.s = s;
+        this.compressed = compressed;
+    }
+
+    public byte[] toBytes() {
+        byte[] rnbufbase = r.toByteArray();
+        byte[] rnbuf = new byte[32];
+        byte[] snbuf = s.toByteArray();
+        //this is basically removing the leading 0 byte, need to test that extensively, did that to be 100% in sync with bitcore.js
+        System.arraycopy(rnbufbase, 1, rnbuf, 0, rnbufbase.length - 1);
+
+        //todo: test this extensively
+        boolean rneg = (rnbuf[0] & 0x80) != 0;
+        boolean sneg = (snbuf[0] & 0x80) != 0;
+
+        byte[] zero = new byte[1];
+        zero[0] = 0x00;
+
+        byte[] drn = new byte[rnbuf.length + 1];
+        System.arraycopy(zero, 0, drn, 0, 1);
+        System.arraycopy(rnbuf, 0, drn, 1, rnbuf.length);
+
+        byte[] dsn = new byte[snbuf.length + 1];
+        System.arraycopy(zero, 0, dsn, 0, 1);
+        System.arraycopy(snbuf, 0, dsn, 1, snbuf.length);
+
+        byte[] rbuf = rneg == true ? drn : rnbuf;
+        byte[] sbuf = sneg == true ? dsn : snbuf;
+
+        int rlength = rbuf.length;
+        int slength = sbuf.length;
+        int len = 2 + rlength + 2 + slength;
+        int rheader = 0x02;
+        int sheader = 0x02;
+        int header = 0x30;
+
+        ByteWriter writer = new ByteWriter(0);
+        writer.putUInt8(header);
+        writer.putUInt8(len);
+        writer.putUInt8(rheader);
+        writer.putUInt8(rlength);
+        writer.putBytes(rbuf);
+        writer.putUInt8(sheader);
+        writer.putUInt8(slength);
+        writer.putBytes(sbuf);
+
+        return writer.toBytes();
+    }
+
+    // DER signature format: [30] [total len] [02] [len R] [R] [02] [len S] [S]
+    // BUG FIX
+    // data is DER format,just use DERSequenceGenerator to handle it.
+    public byte[] toDER() {
+        //return toBytes();
+        return toDER(r, s);
+    }
+    
+    private static byte[] toDER(BigInteger r, BigInteger s) {
+        ByteArrayOutputStream bos = new ByteArrayOutputStream(72);
+        DERSequenceGenerator seq = null;
+        byte[] res = new byte[0];
+        try {
+            seq = new DERSequenceGenerator(bos);
+            seq.addObject(new DERInteger(r));

```

Comment:
yes.
But the toBytes method does not get the correct result,i debug find the problem.
Because the time is long, can not provide detailed data differences.
And I didn't know other way to get the right results.